### DISCUSSION TITLE
Does using docker compose cost extra?

### DISCUSSION THREAD

**Discussion by shmatt** [community/community#59411] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I can't seem to find an answer to this by searching, and Githubs billing breakdown doesn't seem to be fine-grained enough to work it out myself.

When I using a `docker-compose.yml` in my `devcontainer.json`, does codespaces spin up and charge me for the additional containers?  If so, do they use the same container types as my workspace (8 core etc).  Or are the additional containers spun up inside the container type allocation that I have?

**Comment by Fengyi-Li** [comment, marked as answer]
No, using docker compose feature will not cost extra. Codespaces only charges for storage and core hours https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts
